tests/test_pdf.py

```diff
@@ -75,6 +76,25 @@ def test_core_pdf_broadcasting():
     assert broadcasted.shape    == np.array(data).shape
     assert np.all(naive_python  == broadcasted)
 
+def test_add_unknown_modifier():
+    spec = {
+        'channels': [
+            {
+                'name': 'channe',
+                'samples': [
+                    {
+                        'modifiers': [
+                            {'name': 'a_name', 'type': 'this_should_not_exist', 'data': None}
+                        ]
+                    },
+                ]
+            }
+        ]
+    }
+    with pytest.raises(KeyError):
```

Comment:
`pytest.raises(modifiers.InvalidModifier)`

Reply:
good point, in general should we have a `pyhf.exceptions` module?

Reply:
yeah, I was thinking same.

Reply:
done. is probably out of scope for this PR, but good issue